WalletWasabi/Wallets/BlockDownloadStats.cs

```diff
@@ -0,0 +1,153 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using NBitcoin.Protocol;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Wallets;
+
+public class BlockDownloadStats
+{
+	private double AverageLastSample { get; set; }
+	private double StandardDeviationLastSample { get; set; }
+	private int ConsecutiveFails { get; set; }
+	private IEnumerable<BlockDl> Sample { get; set; } = Enumerable.Empty<BlockDl>();
+	private List<BlockDl> History { get; } = new();
+	private object Lock { get; } = new();
+	
+	public void AddBlockDl(IPAddress from, double msElapsed, bool success)
+	{
+		lock (Lock)
+		{
+			History.Add((new BlockDl(from, msElapsed, success)));
```

Comment:
History is just growing infinitely. There are users who run Wasabi for weeks. Looks like a memory leak.